### DISCUSSION TITLE
Suggestions for new features

### DISCUSSION THREAD

**Discussion by archaeodav** [archaeodav/felttegn#24] [Ideas]
### Collated list of proposed new features
- Progress bars. Currently there's no indication for the user that it's working. 
- Stratigraphic indicators- using two points to indicate which overlapping / intersecting feature cuts another. Eventually can feed into Harris Matrix generation via pygraph
- Automatic generation of marker file from coded targets for Metashape
- Custom codes and layer defintiions for different museums. *These are set using the layer_defintion.json, but we need to provide a helper function to make this a bit eaisier- especially saving default symbology*

Via @nesnejretep:
- ~~add average of elevation for anlæg~~ *Done! v1.23*
- run 0m buffer to fix geometry for polygon
- ~~Make UI ask for journalnr og kampagnedato (in addition to opmåler, dato etc.) which is added to the attributes (makes merging layers easier - explicit to which case it is associated)~~ *Done v1.24*

Please add suggestions for features below. Ones to be implemented will be pasted up here for visibility

**Comment by rensefar** [comment]
tilføjelse af TS eller GPS i tabelstrukturen

**Reply by archaeodav** [reply]
Done!

**Comment by rensefar** [comment]
change date format to:  YYYY-MM-DD fx. 2021-10-29

**Reply by archaeodav** [reply]
~Done!~

Realise you mean in the UI and not the table

**Reply by archaeodav** [reply]
Fixed now

**Comment by archaeodav** [comment]
add average of elevation for anlæg

**Reply by archaeodav** [reply]
Done v1.23

**Comment by archaeodav** [comment]
run 0m buffer to fix geometry for polygons

**Comment by nesnejretep** [comment]
Make UI ask for journalnr og kampagnedato (in addition to opmåler, dato etc.) which is added to the attributes (makes merging layers easier - explicit to which case it is associated)

**Reply by archaeodav** [reply]
Sure- Is kampagnedato just the year?

**Reply by nesnejretep** [reply]
Usually
Journal: FHM9999
Kampagnedato: 20211126 (with 0 prefix for days and months between 1 and 9 incl)

**Reply by archaeodav** [reply]
Done in upcoming version 1.24. Kampagne needs some sort of validator

**Comment by karlhjalte** [comment]
Hi

I know you can remove the prefix in the code for features etc, ie. A100 to 100. 

But could it perhaps simply be an option in the selection box when people process data?

Prefix  [+/-]

**Reply by archaeodav** [reply]
I'll get back to you on this with something to test soon

**Reply by karlhjalte** [reply]
any updates on removing prefix? 

It would be beneficial if it is just a number instead of the added "A" to features

**Reply by archaeodav** [reply]
Other museums prefer the prefix- there's a solution in place by editing the layer_definition.json file and replacing prefix with ''. I'm not going to add this to a selection box because I anticipate it causing confusion and inconsistent data.

The intention is rather to maintain a custom configuration for each museum in that file. This will override the default layer naming and styling (also custom feature codes). This functionality should work, but hasn't been tested yet.

If you would like to contibute to the project this would be a great start-  copy the default config  and add one for 'HOM' with your changes and test it, and when it works upload here. I thoroughly recommend GitHub Desktop for managing the changes.

**Reply by karlhjalte** [reply]
ah... sorry. You already mentioned this earlier. Just haven't had the time to look at it. 

"1.2: Functionality to override the default configuration. Currently everything is under the 'default' museum, but you can now add layers, codes or styles in the JSON file under a 3 letter museum code and these will override those in the default"

I will look into it now

**Reply by archaeodav** [reply]
Cool. If you're on windows Notepad++ handles JSON nicely

**Comment by nesnejretep** [comment]
Synopsis: Perhaps add a button in the UI called 'settings' (could be hidden behind the current layout in a tab-view as a separate tab, to not confuse the noob user). The tab could also include very general advice/syntax for edits to the config, and outline which edits are possible.
Action: onclick will open the config JSON in user's standard text editor.
Purpose: locating files in QGIS can be difficult. We want to invite people to do their own customization.

**Reply by archaeodav** [reply]
Good idea- although we need to work out how to apply these on an organisational level when they're distributed through the qgis repo (so that the museum's settings are imported with a fresh install without the user having to worry about it.

Also- while some flexibility is good I'd be wary of enabling too much if we want maintainable, compatible data, because currently that JSON config can be used to set *anything and everything*. We need to ensure that we retain a base-level compatibility with Museum's GIS, MUD etc. I think that's arguably not something we want end users messing about with, and having a slight technical barrier is maybe not a bad thing

I'd much rather do this outwith the qgis user interface- possibly even using a Google form / spreadsheet that we can use to get and set preferences from the various organisations- that way we can pull them from that and not have to worry about the turnaround time to submit and get things approved, and people don't have to mess with git.

**Comment by nesnejretep** [comment]
I have now added progressbar in the new UI. The progress max is set to the number of input files times the number of layers. It gives a fairly reasonable feedback in my tests

**Comment by karlhjalte** [comment]
Add creation of empty data files with same table structure, ie. an empty FUND file without anything in it yet.
